powerlevel9k.zsh-theme

```diff
@@ -973,18 +973,31 @@ build_right_prompt() {
     # Check if it is a custom command, otherwise interpet it as
     # a prompt.
     if [[ $element[0,7] =~ "custom_" ]]; then
-      "prompt_custom" "right" "$index" $element[8,-1]
+      "prompt_custom" "right" "$index" $element[8,-1] >> $POWERLEVEL9K_SOCKET
     else
-      "prompt_$element" "right" "$index"
+      "prompt_$element" "right" "$index" >> $POWERLEVEL9K_SOCKET
     fi
 
     index=$((index + 1))
+    if [[ "$POWERLEVEL9K_RPROMPT_RENDER_ASAP" == true ]]; then
+      kill -s USR1 $$
+    fi
   done
+  if [[ "$POWERLEVEL9K_RPROMPT_RENDER_ASAP" != true ]]; then
+    kill -s USR1 $$
+  fi
 }
 
+if [[ "$POWERLEVEL9K_DISABLE_RPROMPT" != true ]]; then
```

Comment:
Since this is something of initialization code, I think I would prefer to have it up closer to the top of the file rather than down where all of the functions are.

Also, if the user has configured this to print ASAP, is the socket necessary?